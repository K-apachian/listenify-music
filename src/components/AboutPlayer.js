import React, { useState } from "react";
import { Modal, Button, Container, Row, Col, Card } from "react-bootstrap";
import { Link } from "react-router-dom";
import listenmusic from "../images/listenmusic.jpg";
import * as FaIcons from "react-icons/fa";
import "../mystyles/AboutPlayer.css";

const AboutPlayer = ({ ...props }) => {
  const [modalShow, setModalShow] = useState(false);
  return (
    <div>
      <Modal
        {...props}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header style={{ backgroundColor: "crimson" }} closeButton>
          <Modal.Title id="contained-modal-title-vcenter">
            <h2>
              <b>Welcome</b>
            </h2>
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Container>
            <Row>
              <Col style={{ textAlign: "center" }}>
                <h3>Get Connected With Us !</h3>
                <Card className="myaboutcard">
                  <Card.Body>
                    <Row className="mb-5">
                      <h4 className="aboutcard-signup">New User ! Sign Up</h4>
                      <Link to="/register">
                        <Button
                          variant="outline-primary"
                          onClick={props.onHide}
                        >
                          Sign Up
                        </Button>
                      </Link>
                    </Row>
                    <hr style={{ color: "black" }} />
                    <Row className="mb-5">
                      <h4 className="aboutcard-login">Already a User? Sign In</h4>
                      <Link to="/login">
                        <Button variant="outline-dark" onClick={props.onHide}>
                          Sign In
                        </Button>
                      </Link>
                    </Row>
                    <ul class="nav justify-content-center">
                      <a class="nav-link">
                        <i>
                          <FaIcons.FaFacebookF />
                        </i>
                      </a>
                      <a class="nav-link">
                        <i>
                          <FaIcons.FaInstagram />
                        </i>
                      </a>
                      <a class="nav-link">
                        <i>
                          <FaIcons.FaTwitter />
                        </i>
                      </a>
                      <a class="nav-link">
                        <i>
                          <FaIcons.FaYoutube />
                        </i>
                      </a>
                      <a class="nav-link">
                        <i>
                          <FaIcons.FaWhatsapp />
                        </i>
                      </a>
                    </ul>
                  </Card.Body>
                </Card>
              </Col>
              <Col className="aboutcard-image">
                <img
                  src={listenmusic}
                  style={{ width: "100%", height: "100%" }}
                />
              </Col>
            </Row>
          </Container>
        </Modal.Body>
      </Modal>
    </div>
  );
};

export default AboutPlayer;
