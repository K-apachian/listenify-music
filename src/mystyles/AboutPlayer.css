.myaboutcard {
  background: transparent !important;
  border: none;
  color: cyan;
  margin-top: 150px;
  text-align: center;
}

.aboutcard-signup {
  color: midnightblue;
}

.aboutcard-login {
  color: firebrick;
}

.nav-link {
  color: crimson;
}

@media only screen and (max-width: 1000px) {
    .aboutcard-image {
      display: none;
    }
  }