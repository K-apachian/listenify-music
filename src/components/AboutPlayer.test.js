import React from "react";
import { shallow } from "enzyme";
import { expect, it } from "@jest/globals";

import AboutPlayer from "./AboutPlayer";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureMockStore();
const store = mockStore({
  userData: {
    loading: false,
    users: [],
    user: {},
    error: "",
    isLoggedIn: false,
  },
  songsData: {
    loading: false,
    songs: [],
    error: "",
  },
  playlistData: {
    loading: false,
    playlists: [],
    error: "",
  },
});

it("should render entire AboutPlayer Component", () => {
  const wrapper = shallow(
    <Provider store={store}>
      <AboutPlayer />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
