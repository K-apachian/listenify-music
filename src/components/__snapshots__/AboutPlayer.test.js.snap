// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render entire AboutPlayer Component 1`] = `ShallowWrapper {}`;
